### ISSUE TITLE
Import Movable Type from SQLite? Export file?

### ISSUE THREAD

**Issue by sixten** [jekyll/jekyll-import#233]
I have an old blog that's using Movable Type, with the content stored in a SQLite database. Reading the (brief) documentation for the Movable Type importer, it sounds like that only supports MySQL and PostgreSQL databases. Any sense for whether it would be possible to make that work, or how hard it'd be? (My Ruby is rudimentary, at best.)

Failing that, I could use the MT export functionality to dump all of the content to a text file, but it doesn't look like there's an importer for that format. Is that the case?

**Comment by parkr** [comment]
Hey!

> it sounds like that only supports MySQL and PostgreSQL databases. Any sense for whether it would be possible to make that work, or how hard it'd be?

Yes, presently [only allows the mysql and postgresql databases](https://github.com/jekyll/jekyll-import/blob/e0c8b32dcf5fdc4d3a501af71be6e6e8f9f21d81/lib/jekyll-import/importers/mt.rb#L84-L86). That said, it uses the [Sequel gem](http://sequel.jeremyevans.net/) which [supports sqlite](http://sequel.jeremyevans.net/rdoc/files/doc/opening_databases_rdoc.html#label-sqlite). Adding support for it should be as simple as:

- [ ] Add it to allowed engines
- [ ] Ensure database path can be specified via options
- [ ] Run a bunch of tests locally to ensure you get the proper output.

I have added this option in #234. Would you install that version and try again?

**Comment by parkr** [comment]
cc @sixten

**Comment by sixten** [comment]
Sorry, Christmas sort of interrupted my testing and migration efforts. Will try to get back to this on Monday. Really appreciate your quick response to my question!

**Comment by sixten** [comment]
OK, I got your branch installed, and was able to successfully pull the content from my MT database.

I did discover that the updated `mt.rb` had a couple of variable name typos in it from the refactoring. I flagged them on the commit. I fixed those in my local copy of the script, and was able to run it.

I also found that I had to add the `sqlite3` and `sequel` gems to my Gemfile, along with the `jekyll-import` gem before it could actually open my database. That's probably entirely expected, but it wasn't clear to me as a novice Rubyist and Jekyll newb, and might be worth mentioning in the installation docs for the importer.

Once I worked through those things, all of the posts showed up as expected. I have some additional cleanup to do on them, but that's a substantial amount of work saved.
